transaction/core/src/onetime_keys.rs

```diff
@@ -266,17 +242,23 @@ mod tests {
         let tx_private_key = RistrettoPrivate::from_random(&mut rng);
         let (tx_target_key, tx_public_key) = get_output_public_keys(&tx_private_key, &recipient);
 
-        assert!(view_key_matches_output(
-            &account.view_key(), // (a, D_0)
-            &tx_target_key,
-            &tx_public_key
-        ));
+        assert_eq!(
+            recipient.spend_public_key(),
+            &recover_public_subaddress_spend_key(
+                &account.view_private_key(), // (a, D_0)
+                &tx_target_key,
+                &tx_public_key
+            )
+        );
 
         let other_account = AccountKey::random(&mut rng);
-        let bad_view_key = other_account.view_key();
-        assert_eq!(
-            view_key_matches_output(&bad_view_key, &tx_target_key, &tx_public_key),
-            false,
+        assert_ne!(
```

Comment:
It looks like the original test, tried to do the view key matching, but this test is checking the spend key. Why is that?